src/gpl/src/initialPlace.cpp

```diff
@@ -106,23 +142,157 @@ void InitialPlace::doBicgstabPlace() {
 
   // set ExtId for idx reference // easy recovery
   setPlaceInstExtId();
+
+  // Parameters that don't change with iteration
+  int m = pb_->placeInsts().size(); // number of rows of matrix A
+  float tol = 1e-6; // Threshold to decide if the 
+  int reorder = 0;  // "0" for common matrix meaning no extra processing
+  int singularity = -1; // Output. Showing if the sparse matrix is signular or not. -1 = A is invertible
+
   for(int i=1; i<=ipVars_.maxIter; i++) {
     updatePinInfo();
     createSparseMatrix();
+    if (GPU == 1){
```

Comment:
We should move the CUDA code into a seperate method that just takes the sparse values and returns a solution std::vector. Currently the CUDA code is very verbose in what was otherwise a short function.

Reply:
So right. Chris mentioned this as well. I can add a small function outside storing CUDA related stuff. I will use a few lines to call it in this function.